### DISCUSSION TITLE
Can't use extensions assembly built from Netfx.wixext repo with Candle.exe

### DISCUSSION THREAD

**Discussion by MSLukeWest** [wixtoolset/issues#6302] [Questions] [answered]
I'm building the https://github.com/wixtoolset/Netfx.wixext repo (sync'd to master with no local changes) and trying to use the generated assembly WixToolset.Netfx.wixext.dll from the v4 version of Candle like so:

Candle.exe MyExample.wxs -"MyBuiltBinaryPath\WixToolset.Netfx.wixext.dll"

But this doesn't work.  I've tried this with MyExample.wxs using Netfx.wixext in various ways (PropertyRef to "WIX_IS_NETFRAMEWORK_462_OR_LATER_INSTALLED", adding netfx:NativeImage element, etc.), all of which work as expected if I use WixNetFxExtension.dll from the 4.0 toolset.

This repo produces two versions of WixToolset.Netfx.wixext.dll, each of which fail differently:

1. Netfx.wixext\build\Release\netcoreapp3.1\WixToolset.Netfx.wixext.dll - If I use this one I get the same failure I'd get if I don't specify any extension.  Something like "error CNDL0200 : The File element contains an unhandled extension element 'NativeImage'.  Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/netfx' namespace has been provided."
2. Netfx.wixext\build\Release\netstandard2.0\WixToolset.Netfx.wixext.dll - If I use this one I get a reflection exception:

candle.exe : error CNDL0001 : Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.

Exception Type: System.Reflection.ReflectionTypeLoadException

Stack Trace:
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Reflection.Assembly.GetTypes()
   at WixToolset.ExtensionManager.<>c__DisplayClass2_0`1.<Create>b__0(Assembly a)
   at System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at WixToolset.ExtensionManager.Create[T]()
   at WixToolset.Tools.Candle.ParseCommandLineAndLoadExtensions(String[] args)
   at WixToolset.Tools.Candle.Execute(String[] args)

Is there a special way that I need to build and/or consume the output of this repo?

* Which version of WiX are you building with?

> 4.0.0.5918

* Which version of Visual Studio are you building with (if any)?

> VS 2019 Update 8

**Comment by rseanhall** [comment]
@MSLukeWest The WiX Toolset does not use GitHub issues for support. I have converted this issue into a discussion item, which is an experiment that we're trying out. Alternatively, you may want to ask your question on the [wix-users mailing list](https://wixtoolset.org/documentation/mailinglist/) or [Stack Overflow](https://stackoverflow.com/questions/tagged/wix).

candle.exe is v3.x. The Netfx.wixext repo is 4.x. You cannot use a 4.x extension with 3.x.

**Comment by robmen** [comment, marked as answer]
That version of v4 candle.exe is too old to use the .wixext from this repo. More specifically candle.exe was removed from v4 before that point in `master`.

You'll need wix.exe to experiment with the latest .wixexts

**Reply by MSLukeWest** [reply]
Thanks @rseanhall and @robmen for the replies.  I'll make sure to use discussion items for questions from now on.

Can you point me to instructions for getting and using Wix.exe?  I don't see it included in the latest version of the 4.0 toolset.

**Reply by rseanhall** [reply]
Until we [release preview.0](https://www.firegiant.com/blog/2020/6/24/wix-online-meeting-191-highlights/), the way to get wix.exe is to build it from source by running appveyor.cmd from the [Tools repo](https://github.com/wixtoolset/Tools).
